### PR TITLE
Add copyright headers and checks

### LINKED ISSUE #4: Add license
Hi folks,
Please ensure that this repo has a license associated with it, concretely that there is a valid LICENSE file present. Please also consider adding license headers to all files as per [generally accepted best practices](https://softwareengineering.stackexchange.com/questions/317041/should-i-add-the-license-in-every-header-and-source-file/357796).
Thanks

### PR DESCRIPTION
**Pull request by DomFleischmann**
- Add LICENSE file and copyright headers
- Add flake8 checks for headers
- Add tox.ini file for lining

Fixes #4

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -15,14 +15,10 @@ jobs:
 
     - name: Install dependencies
       run: |
-        sudo apt-get install python3-setuptools
-        sudo pip3 install black flake8
+        sudo apt install tox
```
Comment (jnsgruk) [inline .github/workflows/ci.yaml:18]: I'd probably recommend doing `pip install tox` over `apt install`, the version that is packaged in the archives is quite old.

File: test-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Copyright 2021 Canonical Ltd.
```
Comment (jnsgruk) [inline test-requirements.txt:1]: You could inline the contents of this file directly in your tox config?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[flake8]
+max-line-length = 100
+ignore = W503
+copyright-check = True
+copyright-author = Canonical Ltd.
+select = E,F,W,C
```
Comment (jnsgruk) [inline tox.ini:9]: If you wish to be consistent with other teams, consider putting this in a pyproject.toml with other tool configs and leave the tox file to *just* define environments.

Comment (DomFleischmann) [reply tox.ini:9]: Given that we don't have any `pyproject.toml` files in our repositories I prefer having this in our `tox.ini` I also find it a bit more intuitive as this is where flake should be called.

File: test-requirements.txt (hunk 2 above)
Comment (DomFleischmann) [reply test-requirements.txt:1]: I agree, I just did it this way for the sake of consistency across all our repositories.

**Review by DnPlas — APPROVED** [review]

**Pull request merged**